changelogs/unreleased/gh-8783-xlog-garbage-fix.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/core
+
+* Fixed a bug causing garbage being stored in the `stream_id` field of each row
+  header written to an xlog file (gh-8783).
```

Comment:
```suggestion
* Fixed a bug that caused writing incorrect values into the `stream_id` field of xlog headers
  in the debug build (gh-8783).
```

I think it's important to highlight that the bug affected only debug builds.

Reply:
Release builds are affected as well - the bug is just less likely to happen there.

Changed the changelog to

```md
## bugfix/core

* Fixed a bug that caused writing incorrect values into the `stream_id` field
  of xlog headers (gh-8783).
```